src/shim.nim

```diff
@@ -0,0 +1,163 @@
+import c/shim as go_shim
+import types
+
+export SignalCallback
+
+# All procs start with lowercase because the compiler will also need to import
+# status-go, and it will complain of duplication of function names
+
+proc hashMessage*(message: string): string =
+  $go_shim.hashMessage(message.cstring)
```

Comment:
When using this nim-status as a dependency of a nim project, what do you think of instead of returning the string json, return the `"result"` value directly. This should simplify the usage for the library consumers since they would not have to decode the json themselves, and instead they'd receive the hash message string.

Reply:
You should be able to use the not-shimmed `hashMessage` proc via `import nim-status/lib` (not sure if correct syntax).

The procs in `src/shim.nim` are shims around the Nim implementations in `src/lib.nim` intended to preserve the call signatures of status-go. In turn `src/nim_status.nim` builds up a hybrid shim.

Down the road, a broader discussion could be had about possibly dropping the status-go call signatures altogether. But for now it seemed best at the top-level (`src/nim_status.nim` and `src/c/nim_status.nim`) to preserve them.

Reply:
Okay, did an experiment and in `nim-status-client` I can't `import nim_status/status_go` unless `status_go` is moved into a subdirectory named `nim_status`.

So... for `import nim_status/lib` to work (which was/is my intention) I'll need to reorganize a bit — nothing major, just everything inside `src/` (except `src/nim_status.nim`) will need to be moved into `src/nim_status/`, and then I'll need to adjust some relative directories in the sources and tests.